Got nil pointer error when deploying Harbor instance

Got nil pointer error when installing Harbor instance with the following CR:
```
  inClusterStorage:
    kind: MinIO
    minIOSpec:
      # TODO enable redirect
      redirect:
        disable: true
      replicas: 2
...
```
Seems that the PR https://github.com/goharbor/harbor-operator/pull/373 changes the `StorageRedirectSpec.Disable` to `StorageRedirectSpec.Enable`, but anyway the harbor-operator shouldn't crash.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x192e060]

goroutine 664 [running]:
github.com/goharbor/harbor-operator/pkg/cluster/controllers/storage.(*MinIOController).generateMinIOCR(0xc0004ef5f0, 0x2322780, 0xc0002a59c0, 0xc0008a3680, 0xc00025f780, 0xc000c210a0, 0xc000c210a0)
	/home/runner/work/harbor-operator/harbor-operator/pkg/cluster/controllers/storage/provision.go:260 +0x5e0
github.com/goharbor/harbor-operator/pkg/cluster/controllers/storage.(*MinIOController).Apply(0xc0004ef5f0, 0x2322780, 0xc0002a59c0, 0xc0008a3680, 0xc000f2d620, 0x203000, 0x203000)
	/home/runner/work/harbor-operator/harbor-operator/pkg/cluster/controllers/storage/minio.go:84 +0xcb
github.com/goharbor/harbor-operator/controllers/goharbor/harborcluster.(*ServiceManager).Apply(0xc0000536e8, 0x0, 0x0)
	/home/runner/work/harbor-operator/harbor-operator/controllers/goharbor/harborcluster/service_mgr.go:142 +0x1a9
github.com/goharbor/harbor-operator/controllers/goharbor/harborcluster.(*Reconciler).Reconcile.func4(0x0, 0x1)
	/home/runner/work/harbor-operator/harbor-operator/controllers/goharbor/harborcluster/harborcluster.go:104 +0xd5
github.com/goharbor/harbor-operator/pkg/cluster/gos.(*Group).Go.func1(0xc0002a5ac0, 0xc0009afbc0)
	/home/runner/work/harbor-operator/harbor-operator/pkg/cluster/gos/group.go:53 +0x60
created by github.com/goharbor/harbor-operator/pkg/cluster/gos.(*Group).Go
	/home/runner/work/harbor-operator/harbor-operator/pkg/cluster/gos/group.go:49 +0x62
```

--- a/pkg/cluster/controllers/storage/provision.go
+++ b/pkg/cluster/controllers/storage/provision.go
@@ -178,7 +178,7 @@ func (m *MinIOController) Provision() (*lcm.CRStatus, error) {
 func (m *MinIOController) generateIngress() (*netv1.Ingress, error) {
 	var tls []netv1.IngressTLS
 
-	if m.HarborCluster.Spec.InClusterStorage.MinIOSpec.Redirect.Expose.TLS.Enabled() {
+	if m.HarborCluster.Spec.InClusterStorage.MinIOSpec.Redirect.Expose != nil && m.HarborCluster.Spec.InClusterStorage.MinIOSpec.Redirect.Expose.TLS.Enabled() {
 		tls = []netv1.IngressTLS{{
 			SecretName: m.HarborCluster.Spec.InClusterStorage.MinIOSpec.Redirect.Expose.TLS.CertificateRef,
 			Hosts:      []string{m.HarborCluster.Spec.InClusterStorage.MinIOSpec.Redirect.Expose.Ingress.Host},
@@ -237,6 +237,13 @@ func (m *MinIOController) generateMinIOCR(ctx context.Context, harborcluster *go
 		return nil, err
 	}
 
+	externalCertSecret := &minio.LocalCertificateReference{}
+
+	if m.HarborCluster.Spec.InClusterStorage.MinIOSpec.Redirect.Expose != nil && m.HarborCluster.Spec.InClusterStorage.MinIOSpec.Redirect.Expose.TLS != nil {
+		externalCertSecret.Name = m.HarborCluster.Spec.InClusterStorage.MinIOSpec.Redirect.Expose.TLS.CertificateRef
+		externalCertSecret.Type = "kubernetes.io/tls"
+	}
+
 	return &minio.Tenant{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       minio.MinIOCRDResourceKind,
@@ -256,14 +263,11 @@ func (m *MinIOController) generateMinIOCR(ctx context.Context, harborcluster *go
 				Labels:      m.getLabels(),
 				Annotations: m.generateAnnotations(),
 			},
-			ExternalCertSecret: &minio.LocalCertificateReference{
-				Name: m.HarborCluster.Spec.InClusterStorage.MinIOSpec.Redirect.Expose.TLS.CertificateRef,
-				Type: "kubernetes.io/tls",
-			},
-			ServiceName:     m.getServiceName(),
-			Image:           image,
-			ImagePullPolicy: m.getImagePullPolicy(ctx, harborcluster),
-			ImagePullSecret: m.getImagePullSecret(ctx, harborcluster),
+			ExternalCertSecret: externalCertSecret,
+			ServiceName:        m.getServiceName(),
+			Image:              image,
+			ImagePullPolicy:    m.getImagePullPolicy(ctx, harborcluster),
+			ImagePullSecret:    m.getImagePullSecret(ctx, harborcluster),
 			Zones: []minio.Zone{
 				{
 					Name:                DefaultZone,

